File: virtualteachergenaidemo.client/src/models/ScenarioItem.ts
import { VoiceSettings } from './VoiceSettings';
export class ScenarioItem {
    id: string;
    name: string;
    description: string;
    agents: Agent[];
    users: { userId: string, sharedUser: boolean }[];
    voice: VoiceSettings;

    constructor(id: string, name: string, description: string, agents: Agent[], users: { userId: string, sharedUser: boolean }[], voiceSettings: VoiceSettings) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.agents = agents;
        this.users = users;
        this.voice = voiceSettings;
    }
}

// Add class Agent

export class Agent {
    id: string;
    name: string;
    prompt: string;
    type: string;
    features: { feature: string, prompt: string }[] | null | undefined;    
    constructor(id: string, name: string, prompt: string, type: string, features: { feature: string, prompt: string }[] | null | undefined) {
        this.id = id;
        this.name = name;
        this.prompt = prompt;
        this.type = type;
        this.features = features;
     
    }
}

File: VirtualTeacherGenAIDemo.Server/Models/Storage/ScenarioItem.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using VirtualTeacherGenAIDemo.Server.Storage;
using VirtualTeacherGenAIDemo.Server.Options;


namespace VirtualTeacherGenAIDemo.Server.Models.Storage
{
    public class ScenarioItem : IStorageEntity
    {
        public string Id { get; set; } = string.Empty;

        [Required, NotEmptyOrWhitespace]
        public string Name { get; set; } = string.Empty;

        [Required, NotEmptyOrWhitespace]
        public string Description { get; set; } = string.Empty;

        [Required]
        public IEnumerable<Agent>? Agents { get; set; } 

        [JsonIgnore]
        public string Partition => this.Id;
    }

    public class Agent
    {
        [Required, NotEmptyOrWhitespace]
        public string Id { get; set; } = string.Empty;

        [Required, NotEmptyOrWhitespace]
        public string Name { get; set; } = string.Empty;

        [Required, NotEmptyOrWhitespace]
        public string Prompt { get; set; } = string.Empty;

        [JsonIgnore]
        public bool WithFileForSearch { get; set; } = false;

        [Required, NotEmptyOrWhitespace]
        public string Type { get; set; } = string.Empty;

        public List<PromptDashboardFeature> features { get; set; } = new List<PromptDashboardFeature>();


    }
}
